pkg/operator/csi/csicontrollerset/csi_controller_set.go

```diff
@@ -247,18 +247,20 @@ func (c *CSIControllerSet) WithServiceMonitorController(
 func (c *CSIControllerSet) WithStorageClassController(
 	name string,
 	assetFunc resourceapply.AssetFunc,
-	file string,
+	files []string,
 	kubeClient kubernetes.Interface,
 	namespacedInformerFactory informers.SharedInformerFactory,
+	operatorInformer operatorinformer.SharedInformerFactory,
 	hooks ...csistorageclasscontroller.StorageClassHookFunc,
 ) *CSIControllerSet {
 	c.csiStorageclassController = csistorageclasscontroller.NewCSIStorageClassController(
 		name,
 		assetFunc,
-		file,
+		files,
```

Comment:
From this PR itself I could not find a reason while this was changed into an array?

Reply:
Some operators maintain multiple storageclasses, like GCP (see [this PR](https://github.com/openshift/gcp-pd-csi-driver-operator/pull/58)), and the StorageClassState should apply to any SC maintained by that operator.